### ISSUE TITLE
Neotest-zig adapter is used for non-zig repos, resulting in unexpected 'No tests found'

### ISSUE THREAD

**Issue by asfaltboy** [lawrence-laz/neotest-zig#8]
**Describe the bug**

When neotest-zig adapter is enabled, alongside another plugin, such as neotest-go, it's can get selected incorrectly if it's the earlier adapter that is being tested first.

I believe the adapter selection may happen due to the root definition at 

Neotest returns the root if any of the matches are found  ([source](https://github.com/nvim-neotest/neotest/blob/e07fe8241112274aae9947b98d255763738a1d52/lua/neotest/lib/file/init.lua#L374-L376)). And, since  `.git` is a directory that will exist in any git repo, it always matches, which is incorrect.

As far as I understand, neotest should run for `.zig` files even if they aren't detected in a project. So if the root dir does not have any `.zig` files, then the root dir will not be detected as a zig project, but zig tests can still run in individual files (see docs)

## **Steps To Reproduce**
1. Enable two adapters, one of which is `zig`.
2. Start vim and run neotest
3. Receive "No tests found"

## **Current behavior**

Using the zig adapter alongside another neotest adapter takes away the ability to use the other adapter.

## **Expected behavior**

One should be able to use neotest-zig alongside any neotest-* adapter without this unexpected side effect.

## **Logs**

No neotest-zig logs should actually be emitted for this issue, so I didn't bother enabling logging and running those. But, here's the relevant neotest.log excerpt from when neotest ran in a Go test module:

<details><summary>Details</summary>
<p>

```
INFO | 2024-04-02T16:00:14Z+0100 | ...ocal/share/nvim/lazy/neotest/lua/neotest/client/init.lua:380 | Initialising client
INFO | 2024-04-02T16:00:14Z+0100 | ...are/nvim/lazy/neotest/lua/neotest/client/events/init.lua:48 | Emitting starting event
DEBUG | 2024-04-02T16:00:14Z+0100 | ...are/nvim/lazy/neotest/lua/neotest/client/events/init.lua:50 | Calling listener summary for event starting
INFO | 2024-04-02T16:00:14Z+0100 | ...al/share/nvim/lazy/neotest/lua/neotest/adapters/init.lua:18 | Found 1 adapters for directory /home/asfaltboy/Code/learning/gophercises/ex-01-quiz
DEBUG | 2024-04-02T16:00:14Z+0100 | ...al/share/nvim/lazy/neotest/lua/neotest/adapters/init.lua:19 | Adapters: { {
    adapter = {
      _debug_log_path = "/home/asfaltboy/.local/share/nvim/neotest-zig.log",
      _does_file_contain_tests = <function 1>,
      _enable_debug_log = <function 2>,
      _is_debug_log_enabled = false,
      _test_treesitter_query = "\t;;query\n\t(TestDecl\n\t\t[(IDENTIFIER) (STRINGLITERALSINGLE)] @test.name\n\t) @test.definition\n",
      build_spec = <function 3>,
      discover_positions = <function 4>,
      get_strategy_config = <function 5>,
      get_test_node_by_runspec = <function 6>,
      is_test_file = <function 7>,
      name = "neotest-zig",
      results = <function 8>,
      root = <function 9>,
      setup = <function 10>,
      version = "v1.0.5",
      <metatable> = {
        __call = <function 11>
      }
    },
    root = "/home/asfaltboy/Code/learning/gophercises/ex-01-quiz"
  } }
```

</p>
</details> 

## **Environment**
 - OS: ArchLinux
 - Zig version: 0.11.0
 - Nvim version: 0.9.5
 - Neotest version: 5.0.1
 - Neotest Zig version: 742f97f

**Comment by lawrence-laz** [comment]
Hi, thanks for reporting the issue.

I'm currently in the process of rewriting the test runner so it would support latest zig version and would work with both: individual `*.zig` files and modules with external dependencies via `build.zig` among other improvements, like exact test filtering, using single run for all tests, etc.

There's been some issues regarding finding the root already, so I'll need to check that no regressions are introduced and that all works fine with the new test runner.

So the plan is to finish up the rework on test runner first, and then I will try to address this keeping all other things in mind.

**Comment by asfaltboy** [comment]
That sounds great, I appreciate your ongoing dedication to the adapter.

I'm still new to neovim/Lua, so I'd be happy to learn more by reviewing and testing any new changes.

**Comment by lawrence-laz** [comment]
@asfaltboy, I have pushed [a branch `v1.2.0`](https://github.com/lawrence-laz/neotest-zig/tree/v1.2.0), could you test it?
 - It should resolve this issue
 - It requires the latest nightly zig version (don't forget to update `zls` as well, if you use it)
   - Zig v0.12.0 should soon be released, I'll try to release the udpated `neotest-zig` soon after
 - It now supports running tests in projects with `build.zig` (ex. created with `zig init`)

**Comment by asfaltboy** [comment]
It took a while, but looks like arch got zig 0.12 now. But I cannot find the branch you linked to above. Would you be able to push some version that I could test?

EDIT: I see you've pushed the changes fixing the root dir to main, let me give that a go

EDIT 2: I'm getting the error described in #10 , sadly this happens in both go and zig files, so I'm going to disable it for now

**Comment by lawrence-laz** [comment]
Hopefully #10 is now resolved, the author raised a PR. I am on nvim v0.10.0 so only tested on that version.

I also set up `go` with `neotest-go` to test the adapters together. `neotest-zig` doesn't seem to show up as an adapter in `go` repo anymore, and `go` tests seem to execute ok:
<img width="964" alt="image" src="https://github.com/lawrence-laz/neotest-zig/assets/8823448/c351702a-5a1a-4696-a52c-720f59ed9f9a">
